opp_path_selection/src/path_selection_artist.cpp

```diff
@@ -0,0 +1,487 @@
+  opp_msgs::GetPathSelectionCloud srv;
+  pcl::toROSMsg(mesh_cloud, srv.request.input_cloud);
+
+  // now get the path segments from markers and find point indices from mesh along each segment
+  bool success = collectPathPointsCloudCb(srv.request, srv.response);
+  if (!success || !srv.response.success)
+  {
+    message = srv.response.message;
+    return false;
+  }
+  
+  // the response includes res.cloud_indices which represent the path of vertices between each segment of markers/points
+  points_idx.clear();
+  for(int i=0; i<srv.response.cloud_indices.size(); i++)
+    {
+      int idx = srv.response.cloud_indices[i];
+      points_idx.push_back(idx);
+    }
+  
+  if(points_idx.size()>2) return true;
+  ROS_ERROR("No path points found");
+  return false;
+}
+
+bool PathSelectionArtist::collectPathMesh(const shape_msgs::Mesh& mesh_msg,
+				      std::vector<int>& points_idx,
+				      std::string& message)
+{
+  // first get the mesh
+  opp_msgs::GetPathSelectionMesh srv;
+  srv.request.input_mesh = mesh_msg;
+
+  // now get the path segments from markers and find point indices from mesh along each segment
+  bool success = collectPathPointsMeshCb(srv.request, srv.response);
+  if (!success || !srv.response.success)
+  {
+    message = srv.response.message;
+    return false;
+  }
+  
+  // the response includes res.cloud_indices which represent the path of vertices between each segment of markers/points
+  points_idx.clear();
+  for(int i=0; i<srv.response.mesh_indices.size(); i++)
+    {
+      int idx = srv.response.mesh_indices[i];
+      points_idx.push_back(idx);
+    }
+  
+  if(points_idx.size()>2) return true;
+  ROS_ERROR("No path points found");
+  return false;
+}
+
+  
+bool PathSelectionArtist::collectPathPointsCloudCb(opp_msgs::GetPathSelectionCloudRequest& req, opp_msgs::GetPathSelectionCloudResponse& res)
+{
+  // get points from existing marker_array_
+  auto points_it = std::find_if(marker_array_.markers.begin(),
+                                marker_array_.markers.end(),
+                                [](const visualization_msgs::Marker& marker) { return marker.id == 0; });
+
+  // Convert the selection points to Eigen vectors
+  std::vector<Eigen::Vector3d> points;
+  for (const geometry_msgs::Point& pt : points_it->points)
+  {
+    Eigen::Vector3d e;
+    tf::pointMsgToEigen(pt, e);
+    points.push_back(std::move(e));
+  }
+
+  // get point cloud from service request
+  pcl::PointCloud<pcl::PointXYZ>::Ptr cloud(new pcl::PointCloud<pcl::PointXYZ>());
+  pcl::fromROSMsg(req.input_cloud, *cloud);
+
+  // get parameters from file. Note: This file is specified on construction, re-read allows dynamic changes 
+  PathSelectorParameters params;
+  bool success = opp_msgs_serialization::deserialize(path_selection_config_file, params);
+  if (!success)
+  {
+    ROS_ERROR_STREAM("Could not load path selection config from: " << path_selection_config_file);
+    return false;
+  }
```

Comment:
This isn't really a great practice. If you want this kind of dynamic behavior, these parameters should be part of the service request